src/imitation/scripts/train_imitation.py

```diff
@@ -140,6 +141,41 @@ def dagger(
     return stats
 
 
+@train_imitation_ex.command
+def sqil(
+    sqil: Mapping[str, Any],
+    _run,
+    _rnd: np.random.Generator,
+) -> Mapping[str, Mapping[str, float]]:
+    custom_logger, log_dir = logging_ingredient.setup_logging()
+    expert_trajs = demonstrations.get_expert_trajectories()
+
+    with environment.make_venv() as venv:
+        sqil_trainer = SQIL(
+            venv=venv,
+            demonstrations=expert_trajs,
+            policy=sqil["policy_model"],
```

Comment:
This duplicates the policy ingredient: https://github.com/HumanCompatibleAI/imitation/blob/master/src/imitation/scripts/ingredients/policy.py

I think can replace with `policy.make_policy(venv)` and remove the `policy_model` config parameter.

Reply:
> I think can replace with policy.make_policy(venv) and remove the policy_model config parameter.
This wouldn't work because `policy.make_policy(venv)` returns `policies.BasePolicy`, whereas SQIL requires `type[policies.BasePolicy]` (i.e. a constructor, not an instance). 

You are right though that I could use `policy["policy_cls"]`. However, SQIL uses DQN by default which requires a DQNPolicy and the default for `policy_cls` is `base.FeedForward32Policy`. If the default argument doesn't work for SQIL, his would mean that we couldn't run `train_imitation` with the `sqil` command without modifying the policy in the command line. 

One fix would be to change the name of `policy` to `imitation_policy` and `sqil["policy_model"]` to `sqil["sqil_policy_cls"]`. I'll go with this for now, but if you think we should do it differently (e.g. just use policy_model["policy_cls"] for both) I can roll back the changes.

Reply:
> This wouldn't work because policy.make_policy(venv) returns policies.BasePolicy, whereas SQIL requires type[policies.BasePolicy] (i.e. a constructor, not an instance).

Ah, good point, yes the difference between SQIL expecting classes v.s. the rest of our code expecting objects bites again.

> But you are right, I could use policy["policy_cls"]. One issue is that SQIL uses a DQN by, default which requires a DQNPolicy and the default for policy_cls is base.FeedForward32Policy, which is incompatible. This is bit unfortunate, but I can't think of a way around it currently. To make it easy I've made a named_config called sqil.dqn which lets users set a dqn policy.

> I currently can't think of a way to make sqil work by default. We would somehow want to override a config conditional on the sqil command being used. Not sure if that's possible/desirable.

Mm indeed this is messy and highlights a design flaw in Sacred. It is possible to have a different default depending on the context using a config hook, https://github.com/HumanCompatibleAI/imitation/blob/master/src/imitation/scripts/ingredients/rl.py#L41 is an example of this. You can add a hook in SQIL that checks if `policy_cls == base.FeedForward32Policy` and if so changes it to a `DQNPolicy`. This is a bit nasty since if for some reason the user *manually* sets the policy to a FeedForward32Policy we'll just override that silently, but it seems OK.

Reply:
This seems good. I've made the change.